aws/service.go

```diff
@@ -315,6 +316,14 @@ func ApplicationAutoScalingClient(ctx context.Context, d *plugin.QueryData) (*ap
 	return applicationautoscaling.NewFromConfig(*cfg), nil
 }
 
+func ApplicationSignalsClient(ctx context.Context, d *plugin.QueryData) (*applicationsignals.Client, error) {
+	cfg, err := getClientForQueryRegion(ctx, d)
```

Comment:
Could we use the function `getClientForQuerySupportedRegion` instead?

Also please include the following check, it will help us with the unsupported region check.

```go
if cfg == nil {
		return nil, nil
	}
```